### PR TITLE
mon: calculate reporters_by_subtree when adjust_heartbeat_grace is false

### PR DESCRIPTION
**Pull request by wonzhq**
In the case mon_osd_adjust_heartbeat_grace is set to false, should
also calculate reporters_by_subtree.

### CODE REVIEW

**Comment by liewegas** [comment]
It seems like this shouldb e possible without cut and pasting the code.. can you refactor so there aren't two copies of the loop that builds the map?

**Comment by wonzhq** [comment]
I was thinking to save some time to avoid looping the map `fi.reporters` twice in the case when mon_osd_adjust_heartbeat_grace is true. Does this matter? @liewegas

**Comment by wonzhq** [comment]
Updated, pls take another look! @liewegas

**Comment by athanatos** [comment]
@wonzhq Can you rebase?

**Comment by wonzhq** [comment]
Rebased.

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -1777,6 +1777,7 @@ bool OSDMonitor::check_failure(utime_t now, int target_osd, failure_info_t& fi)
 	   << " + " << peer_grace << "), max_failed_since " << max_failed_since
 	   << dendl;
 
+
```
Comment (athanatos) [inline src/mon/OSDMonitor.cc]: @wonzhq This doesn't look right.  Did something go wrong with the rebase?

**Comment by wonzhq** [comment]
Oops, the change in the second commit has been in the master. I've removed it from this PR.

**Comment by tchaikov** [comment]
@wonzhq #8749 fixes the same issue in a different way, @Aran85 keeps just a single loop over `fi.reporters`. personally, i am good with both approaches. but seems @liewegas prefers the "single" loop one. so i am closing yours.

**Pull request closed without merging**